### PR TITLE
Improved settings dialog

### PR DESCRIPTION
**Pull request by RevZero**
See commit logs.

### CODE REVIEW

File: xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ namespace XCOM2Launcher.Forms
     public partial class SettingsDialog : Form
     {
         protected Settings Settings { get; set; }
+        private readonly List<string> ModPaths;
```
Comment (Xymanek) [inline xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs]: Why is this needed? It seems to just follow `modPathsListbox.Items`

Comment (RevZero) [reply xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs]: True. But ListBox.ObjectCollection is a pain to work with. I prefer to avoid Items.Cast<string>().ToList() stuff all the time. Using a second list just seemed to be the lesser evil to me in this case.

Comment (Xymanek) [reply xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs]: Ah I see your point. Could you make separate helper functions for adding/removing entries then that will be responsible for keeping the 2 in sync?

Also, please document your reasoning in the code (probably a comment next to the field declaration)

Comment (RevZero) [reply xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs]: I decided to removed the extra list after all. It wasn't worth keeping it to save two casts.

File: xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs [hunk 2]
```diff
@@ -129,6 +127,8 @@ private void bOK_Click(object sender, EventArgs e)
             Settings.CheckForPreReleaseUpdates = checkForPreReleaseUpdates.Checked;
             GlobalSettings.Instance.IsSentryEnabled = useSentry.Checked;
             Settings.AllowMultipleInstances = allowMutipleInstances.Checked;
+            Settings.GamePath = gamePathTextBox.Text;
+            Settings.ModPaths = modPathsListbox.Items.Cast<string>().ToList();
```
Comment (Xymanek) [inline xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs]: You should probably save `modPathsListbox.Items.Cast<string>().ToList()` to a variable since you do it twice in the same function. Otherwise looks good
